vendor_server.cc

```diff
@@ -39,49 +25,48 @@ using grpc::ServerBuilder;
 using grpc::ServerContext;
 using grpc::Status;
 using grpc::StatusCode;
-using supplyfinder::HelloRequest;
-using supplyfinder::HelloReply;
-using supplyfinder::Vendor;
 using supplyfinder::FoodID;
 using supplyfinder::InventoryInfo;
+using supplyfinder::Vendor;
 
 // Logic and data behind the server's behavior.
 class VendorServiceImpl final : public Vendor::Service {
-  public:
-  VendorServiceImpl () {
+ public:
+  VendorServiceImpl() {
     /*
-     * Dummy database which contains :
-     * id 1: price 19.5, quantity 10
-     * id 3: price 2.99, quantity 2
+     * Randomly generate inventory information
+     * Each vendor has five items, with price [0, 20.0] and quantity [0, 99]
      */
+    std::vector<int> indices = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    std::random_device rd;
+    std::mt19937 g(rd());
+    std::shuffle(indices.begin(), indices.end(), g);
     InventoryInfo inv;
-    inv.set_price(19.5);
-    inv.set_quantity(10);
-    inventory_db[1] = inv;
-    inv.set_price(2.99);
-    inv.set_quantity(2);
-    inventory_db[3] = inv;
-  }
-  Status SayHello(ServerContext* context, const HelloRequest* request,
-                  HelloReply* reply) override {
-    std::string prefix("Hello ");
-    reply->set_message(prefix + request->name());
-    return Status::OK;
+    for (int i = 0; i < 5; i++) {
+      int idx = indices[i];
+      double price = rand() % 200 / 10.0;
+      uint32_t quantity = rand() % 100;
+      inv.set_price(price);
+      inv.set_quantity(quantity);
+      inventory_db_[idx] = inv;
+    }
   }
-  
-  Status CheckInventory (ServerContext* context, const FoodID* request,
-                  InventoryInfo* info) {
+
+  Status CheckInventory(ServerContext* context, const FoodID* request,
+                        InventoryInfo* info) {
     uint32_t food_id = request->food_id();
-    if (inventory_db.find(food_id) == inventory_db.end()) {
+    auto inventory = inventory_db_.find(food_id);
+    if (inventory == inventory_db_.end()) {
       Status status(StatusCode::NOT_FOUND, "Food ID not found.");
       return status;
     }
-    info->set_price(inventory_db[food_id].price());
-    info->set_quantity(inventory_db[food_id].quantity());
+    info->set_price(inventory->second.price());
+    info->set_quantity(inventory->second.quantity());
     return Status::OK;
   }
-  private:
-  std::unordered_map<uint32_t, InventoryInfo> inventory_db;
+
+ private:
+  std::unordered_map<uint32_t, InventoryInfo> inventory_db_;
```

Comment:
It might be worth commenting on the key value being food ID, as it's not immediately apparent to a reader.